webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/internal/NodeDetailsAPI.scala

```diff
@@ -0,0 +1,149 @@
+import com.normation.rudder.domain.policies.PolicyModeOverrides.Unoverridable
+import com.normation.rudder.reports.execution.AgentRunWithNodeConfig
+import com.normation.rudder.reports.execution.RoReportsExecutionRepository
+import com.normation.rudder.repository.json.DataExtractor.OptionnalJson
+import com.normation.rudder.rest.RestExtractorService
+import com.normation.rudder.services.nodes.NodeInfoService
+import com.normation.rudder.web.components.DateFormaterService
+import com.typesafe.config.ConfigRenderOptions
+import net.liftweb.common.Box
+import net.liftweb.common.Full
+import net.liftweb.common.Loggable
+import net.liftweb.http.JsonResponse
+import net.liftweb.http.LiftResponse
+import net.liftweb.http.Req
+import net.liftweb.http.rest.RestHelper
+import net.liftweb.json.JsonAST.JArray
+import net.liftweb.json.JsonAST.JField
+import net.liftweb.json.JsonAST.JNothing
+import net.liftweb.json.JsonAST.JObject
+import net.liftweb.json.JsonAST.JString
+import net.liftweb.json.parse
+import zio.ZIO
+
+class NodeDetailsAPI (
+  nodeInfoService: NodeInfoService
+  , restExtractor : RestExtractorService
+  , reportsExecutionRepository: RoReportsExecutionRepository
+  , getGlobalMode : () => Box[GlobalPolicyMode]
+  , readOnlySoftwareDAO: ReadOnlySoftwareDAO
+
+) extends  RestHelper with  Loggable {
+
+
+  def serialize(agentRunWithNodeConfig: Option[AgentRunWithNodeConfig], globalPolicyMode: GlobalPolicyMode, nodeInfo : NodeInfo, properties : List[String], softs: List[Software]) = {
+    import net.liftweb.json.JsonDSL._
+
+    val (policyMode,explanation) =
+      (globalPolicyMode.overridable,nodeInfo.policyMode) match {
+        case (Always,Some(mode)) =>
+          (mode,"<p>This mode is an override applied to this node. You can change it in the <i><b>node's settings</b></i>.</p>")
+        case (Always,None) =>
+          val expl = """<p>This mode is the globally defined default. You can change it in <i><b>settings</b></i>.</p><p>You can also override it on this node in the <i><b>node's settings</b></i>.</p>"""
+          (globalPolicyMode.mode, expl)
+        case (Unoverridable,_) =>
+          (globalPolicyMode.mode, "<p>This mode is the globally defined default. You can change it in <i><b>Settings</b></i>.</p>")
+      }
+    (  ("name" -> nodeInfo.hostname)
+      ~  ("policyServerId" -> nodeInfo.policyServerId.value)
+      ~  ("policyMode" -> policyMode.name)
+      ~  ("explanation" -> explanation)
+      ~  ("kernel" -> nodeInfo.osDetails.kernelVersion.value)
+      ~  ("agentVersion" -> nodeInfo.agentsName.headOption.flatMap(_.version.map(_.value)))
+      ~  ("id" -> nodeInfo.id.value)
+      ~  ("ram" -> nodeInfo.ram.map(_.toStringMo()))
+      ~  ("machineType" -> nodeInfo.machine.map(_.machineType.toString))
+      ~  ("os" -> nodeInfo.osDetails.fullName)
+      ~  ("state" -> nodeInfo.state.name)
+      ~  ("ipAddresses" -> nodeInfo.ips.filter(ip => ip != "127.0.0.1" && ip != "0:0:0:0:0:0:0:1"))
```

Comment:
there is a lot of other way to write localhost in ipv6, most common is `::1`. 
I'm not sure why you filter them ? It may be important, even if just to see if ipv4/ipv6 is supported.

Reply:
they are never filtering it in the web interface, which should it be filtered here?